### ISSUE TITLE
Uncaught TypeError: Cannot read property 'length' of undefined

### ISSUE THREAD

**Issue by adesurirey** [samu/blocky#22]
[Enter steps to reproduce:]

1. Update Atom to 1.32.0
2. Open a ruby file, write some lines and hit save, the error is raised at every save hit

**Atom**: 1.32.0 x64
**Electron**: 2.0.9
**OS**: Mac OS X 10.13.6
**Thrown From**: [blocky](https://github.com/samu/blocky) package 0.11.0

### Stack Trace

Uncaught TypeError: Cannot read property 'length' of undefined

```
At /Users/Arnaud/.atom/packages/blocky/lib/blockmap-compiler.coffee:87

TypeError: Cannot read property 'length' of undefined
    at module.exports (/packages/blocky/lib/blockmap-compiler.coffee:87:3)
    at BlockyView.module.exports.BlockyView.notifyContentsModified (/packages/blocky/lib/blocky-view.coffee:31:17)
    at /packages/blocky/lib/blocky-view.coffee:11:54
    at Function.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:1185469)
    at Emitter.emit (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:1186910)
    at TextBuffer.emitDidStopChangingEvent (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:500334)
    at later (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:14:98316)
```

### Commands

```
     -0:37.6.0 core:save (input.hidden-input)
     -0:05.3.0 emmet:toggle-comment (input.hidden-input)
     -0:05.3.0 editor:toggle-line-comments (input.hidden-input)
```

### Non-Core Packages

```
atom-beautify 0.33.4 
auto-detect-indentation 1.3.0 
autoclose-html 0.23.0 
autocomplete-paths 2.12.2 
blocky 0.11.0 
busy-signal 1.4.3 
clipboard-plus 0.5.1 
color-picker 2.3.0 
emmet 2.4.3 
erb-snippets 1.1.0 
file-icons 2.1.25 
git-time-machine 2.1.0 
highlight-selected 0.14.0 
intentions 1.1.5 
language-babel 2.85.0 
language-v0 0.4.1 
linter 2.2.0 
linter-erb 1.1.1 
linter-eslint 8.4.1 
linter-js-yaml 1.3.0 
linter-jshint 3.1.9 
linter-ruby 1.2.6 
linter-sass-lint 1.8.3 
linter-scss-lint 3.1.1 
linter-stylelint 4.3.2 
linter-ui-default 1.7.1 
minimap 4.29.9 
pigments 0.40.2 
pretty-json 2.0.3 
project-manager 3.3.6 
rails-i18n-autocomplete 0.4.0 
split-diff 1.5.3 
tabs-closer 0.16.0 
toggle-quotes 1.1.1 
trailing-spaces 0.4.0 
xml-formatter 0.11.0 
```

**Comment by samu** [comment]
This is related to [tree sitter being enabled by default](https://github.com/atom/atom/pull/17879). You can 'fix' the problem by disabling `Use Tree Sitter Parsers` on the Core Settings Page. There will still be a small bug: the token underneath the cursor isn't highlighted anymore.

The background of this issue is that the blocky package was using private Atom API (`editor.tokenizedBuffer.tokenizedLines`). I will have to do further analysis to see if the package can be kept alive.

**Comment by AlexWayfer** [comment]
> if the package can be kept alive

I hope so. :raised_hands: 

It's a cool package, I wouldn't want to lose it.

**Comment by samu** [comment]
This should be fixed with `v0.13.0`.

**Comment by TSMMark** [comment]
Can confirm 0.13.0 has fixed it, thanks @samu!
